From b2adaca92c63b9bb8beb021d554f656e387a7648 Mon Sep 17 00:00:00 2001
From: Joe Perches <joe@perches.com>
Date: Sun, 3 Feb 2013 17:43:58 +0000
Subject: [PATCH v2 006/021] ethernet: Remove unnecessary alloc/OOM messages, alloc
 cleanups

Git-commit: b2adaca92c63b9bb8beb021d554f656e387a7648
Patch-mainline: v3.9-rc1
Reference: fate#317398
Target: sle11-sp4

alloc failures already get standardized OOM
messages and a dump_stack.

Convert kzalloc's with multiplies to kcalloc.
Convert kmalloc's with multiplies to kmalloc_array.
Fix a few whitespace defects.
Convert a constant 6 to ETH_ALEN.
Use parentheses around sizeof.
Convert vmalloc/memset to vzalloc.
Remove now unused size variables.

Signed-off-by: Joe Perches <joe@perches.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Lee, Chun-Yi <jlee@suse.com>

---
 drivers/net/ethernet/intel/igbvf/netdev.c |    8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

Index: linux-3.0-SLE11-SP4/drivers/net/ethernet/intel/igbvf/netdev.c
===================================================================
--- linux-3.0-SLE11-SP4.orig/drivers/net/ethernet/intel/igbvf/netdev.c
+++ linux-3.0-SLE11-SP4/drivers/net/ethernet/intel/igbvf/netdev.c
@@ -1394,12 +1394,10 @@ static void igbvf_set_multi(struct net_d
 	int i;
 
 	if (!netdev_mc_empty(netdev)) {
-		mta_list = kmalloc(netdev_mc_count(netdev) * 6, GFP_ATOMIC);
-		if (!mta_list) {
-			dev_err(&adapter->pdev->dev,
-			        "failed to allocate multicast filter list\n");
+		mta_list = kmalloc_array(netdev_mc_count(netdev), ETH_ALEN,
+					 GFP_ATOMIC);
+		if (!mta_list)
 			return;
-		}
 	}
 
 	/* prepare a packed array of only addresses. */
